plugins/env/lib/samson_zendesk/samson_plugin.rb

```diff
@@ -0,0 +1,41 @@
+module SamsonEnv
+  class Engine < Rails::Engine
+  end
+end
+
+Samson::Hooks.view :stage_form, "samson_env/fields"
+
+Samson::Hooks.callback :stage_permitted_params do
+  :env
+end
+
+# TODO pass arrays instead of calling twice
+Samson::Hooks.callback :stage_permitted_params do
+  {:environment_variable_group_ids => []}
+end
+
+Samson::Hooks.callback :after_deploy_setup do |dir, stage|
+  if stage.env.present? || stage.environment_variable_groups.any?
+    # TODO also include everything here / write multiple of these
+    File.write("#{dir}/.env", stage.env)
+
+    # propriatary manifest format
+    manifest = "#{dir}/ENV.json"
```

Comment:
Do we want to force people to declare a ENV.json manifest file? Maybe leave it up to our process to encourage it, and if it's not there just write the .env files anyway. I would like that as a user if I'm still in the process of creating a project/stage setup in Samson and haven't nailed down all the details yet. Otherwise I'd have to keep doing PRs in the github repo while twiddling settings.